core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```
@@ -126,7 +128,24 @@ object ConsumerGroupCommand extends Logging {
     private var consumer: KafkaConsumer[String, String] = _
 
     def listGroups(): List[String] = {
-      adminClient.listAllConsumerGroupsFlattened().map(_.groupId)
+      val result = adminClient.listConsumerGroups(
+        withTimeoutMs(new ListConsumerGroupsOptions))
+
+      try {
+        val listings = result.all.get.asScala
+        listings.map(_.groupId).toList
+      }
+      catch {
+        case e : ExecutionException if e.getCause.isInstanceOf[KafkaException] =>
+          val kafkaException = e.getCause
+          if (kafkaException.getCause != null) {
```

Seems a little arbitrary to go one additional level. I'd suggest we just raise the cause of the ExecutionException.

By the way, this seems to be the only case where we worry about the ExecutionException. For other admin APIs, we just let it propagate. It would be nice to be consistent if possible. Going forward, I think we expect retriable errors to be handled internally in the AdminClient, which means only fatal errors and timeouts will reach this level. It may be useful to catch the timeouts and print a helpful message, but for anything else, we probably just want to raise the exception and fail the command. At least that seems like a conservative option until we've had a chance to refine the expected AdminClient error handling. So maybe we should remove this `try/catch`. What do you think?

---

Thanks for the clarification, I removed mentioned ``try-catch`` blocks. In general, I find it easier to read helpful error messages than stack traces, but you are right; ``KafkaAdminClient`` class has ``@InterfaceStability.Evolving`` annotation. Error handling can be refined in a separate PR.